panther_manager/include/panther_manager/plugins/action/joy_subscription_node.hpp

```diff
@@ -0,0 +1,55 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef PANTHER_MANAGER_PLUGINS_JOY_CONDITION_NODE_HPP_
+#define PANTHER_MANAGER_PLUGINS_JOY_CONDITION_NODE_HPP_
+
+#include <memory>
+#include <mutex>
+#include <string>
+
+#include <behaviortree_ros2/bt_topic_sub_node.hpp>
+#include <rclcpp/rclcpp.hpp>
+
+#include <sensor_msgs/msg/joy.hpp>
+
+#include "panther_manager/behavior_tree_utils.hpp"
+
+namespace panther_manager
+{
+
+class JoySubscription : public BT::RosTopicSubNode<sensor_msgs::msg::Joy>
```

Comment:
- The class name should correlate with the file name
- No sense to name an add "condition" part if a file is defined in the condition subdirectory
- Name should describe the node responsibility -> `CheckJoyButtons` `AreButtonsPressed` etc.